src/test/kotlin/camp/nextstep/edu/racingcar/racing/CarRaceEventTest.kt

```diff
@@ -0,0 +1,50 @@
+package camp.nextstep.edu.racingcar.racing
+
+import io.kotest.core.spec.style.BehaviorSpec
+import org.assertj.core.api.Assertions.assertThat
+
+class CarRaceEventTest : BehaviorSpec({
+    given("자동차 엔진이 1번 움직일 수 있을 때 ") {
+        val moveOnceEngine = InstantEngine(1)
+        val car = Car(1, "once", moveOnceEngine)
+
+        `when`("자동차가 5번 달리면 ") {
+            val events = mutableListOf<CarRaceEvent>()
+
+            events.add(car.race())
+            events.add(car.race())
+            events.add(car.race())
+            events.add(car.race())
+            events.add(car.race())
```

Comment:
```suggestion
            val events = listOf(TestingCarRaceEvent(moved=true), ...)
```
테스트가 너무 많은 객체를 의존하여 만들어지면 깨지기 쉬워서 유지보수하는데도 좋지 않습니다.
`Car` 객체를 통해 `CarRaceEvent` 를 만들지 말고 직접 설정해도 좋을것 같아요. 이렇게 되면 테스트 안되는 부분이 있을텐데 만약 있다면 그 부분은 별도의 테스트클래스에서 검증할 수 있도록 해보면 좋겠습니다.

추가로 이렇게 바꾸려면 `CarRaceEvent` 클래스의 변경이 좀 필요한데요, 객체 생성이 매우 엄격한 조건에서 한정적으로만 되어 있는데 개인적으로는 여러 방식으로 객체생성을 지원해줘도 되지 않을까 합니다.

```kotlin
class CarRaceEvent(val eventId: Long = idGenerator.getAndIncrement(), val carId: Int, val moved: Boolean) {

    companion object {
        private val idGenerator = AtomicLong(0L)

        fun movedEvent(carId: Int) = CarRaceEvent(carId = carId, moved = true)

        fun stoppedEvent(carId: Int) = CarRaceEvent(carId = carId, moved = false)

        fun merge(events: List<CarRaceEvent>): Track {
            ...
        }
    }
}
```